pkg/master/master.go

```diff
@@ -206,18 +204,19 @@ func (c *Config) createLeaseReconciler() reconcilers.EndpointReconciler {
 	if err != nil {
 		glog.Fatalf("Error creating storage factory: %v", err)
 	}
-	endpointConfig, err := c.ExtraConfig.StorageFactory.NewConfig(kapi.Resource(DefaultStorageEndpoint))
+	endpointPrefix := c.ExtraConfig.StorageFactory.ResourcePrefix(api.Resource("endpoints"))
+	endpointConfig, err := c.ExtraConfig.StorageFactory.NewConfig(kapi.Resource(endpointPrefix))
 	if err != nil {
 		glog.Fatalf("Error getting storage config: %v", err)
 	}
 	endpointsStorage := endpointsstorage.NewREST(generic.RESTOptions{
 		StorageConfig:           endpointConfig,
 		Decorator:               generic.UndecoratedStorage,
 		DeleteCollectionWorkers: 0,
-		ResourcePrefix:          c.ExtraConfig.StorageFactory.ResourcePrefix(kapi.Resource(DefaultStorageEndpoint)),
+		ResourcePrefix:          c.ExtraConfig.StorageFactory.ResourcePrefix(kapi.Resource(endpointPrefix)),
 	})
 	endpointRegistry := endpoint.NewRegistry(endpointsStorage)
-	masterLeases := reconcilers.NewLeases(leaseStorage, "/masterleases/", ttl)
+	masterLeases := reconcilers.NewLeases(leaseStorage, "/masterleases", ttl)
```

Comment:
do `path.Join` internally to ensure we normalize joining with or without leading/trailing slashes?

Reply:
also, does `List` depend on a trailing slash? does removing this mean we could see things at `/masterleases1`?

Reply:
Internally the prefix is appended with a / and a key for specific keys or just the prefix, so shouldn't have an issue with /masterleases1.  It is currently hard-coded so "masterleases/" would result in "masterleases//" on lookup.  Using path.join is the better way to handle the filename.

Reply:
> or just the prefix, so shouldn't have an issue with /masterleases1

my question was whether `List("/masterleases")` would return `/masterleases1`

Reply:
Updated the internals to use path.join when hunting for a key

Reply:
> my question was whether List("/masterleases") would return /masterleases1 if it existed

I believe it would.  The list function is just using a prefix.  In my experience List("masterleases") would return /masterleses1 and/or /masterleases/1

Reply:
so this needs to change back to "/masterleases/"

Reply:
You're right.  I'll fix